src/main_window.cpp

```diff
@@ -264,19 +264,51 @@ void MainWindow::installPkg()
     // Create game directory.
     auto gamesDirectory = readGamesDirectorySetting();
 
-    if (!QDir(gamesDirectory).mkdir(param.titleId())) {
-        QString msg(
-            "Cannot create directory %1 inside %2. "
-            "If you have a failed installation from a previous attempt, you will need to remove this directory before trying again.");
+    // Get Param information
+    auto appver = param.appver();
+    auto category = param.category();
+    auto shortContentId = param.shortContentId();
+    auto title = param.title();
+    auto titleId = param.titleId();
+
+    // Check if file is Patch/DLC.
+    bool patchOrDlc = false;
+    if (category.startsWith("gp") || category.contains("ac")) {
+        patchOrDlc = true;
+    }
+
+    // If PKG isn't a game, DLC, or Patch, don't allow.
+    if (!patchOrDlc && !category.startsWith("gd")) {
+            QString msg("PKG file is not a Patch, DLC, or a Game. Possibly a corrupted PKG?");
+
+            QMessageBox::critical(&progress, "Error (Not Game, Patch, or DLC)", msg);
+            return;
+    }
+
+    if (!QDir(gamesDirectory).mkdir(titleId)) {
```

Comment:
This will return `false` when the user trying to install the patch while the game is already installed.

Reply:
...No it won't? Because the update is named as:

TITLE_ID-1,01
TITLE_ID-1.02

etc.

Reply:
`titleId` here contains only title ID like `CUSAXXXXX`.

Reply:
...Damnit XD